sophia/src/term/iri.rs

```diff
@@ -331,6 +335,28 @@ where
         }
     }
 
+    /// Parses the components of the IRI.
+    ///
+    /// This is necessary to use the IRI as a base to resolve other IRIs.
+    ///
+    /// # Errors
+    ///
+    /// As a regular expression is applied to the internal representation of
+    /// the IRI, it is required that the IRI is not suffixed. Otherwise
+    /// `TermError::IriParse` is raised.
```

Comment:
If prefix ends with a gen-delims, it should in theory be possible to build a consistent IriParsed pointing to parts of the prefix and the suffix. This is not technically trivial though, and should not be our priority here. But maybe worth submitting an issue (tagged with 'improvement')?

Reply:
Yes, I think it is worth an issue, too. Given the current structure of `IriParsed` this is really challenging. I'm not happy the the current solution either as it should work with every IRI (reference). As you already noticed I only try to get the basic stuff working in this PR and leave space for optimizations later.